src/main/java/com/zufar/icedlatte/user/endpoint/UserEndpoint.java

```diff
@@ -120,4 +127,28 @@ public ResponseEntity<Void> deleteUserAvatar() {
         log.info("The request to delete the user avatar was handled.");
         return ResponseEntity.status(HttpStatus.OK).build();
     }
+
+    @Override
+    @PostMapping(path = "/password/reset")
+    public ResponseEntity<Void> resetUserPassword() {
+        UserDto userDto = securityPrincipalProvider.get();
+        log.info("Received the request to reset password the User with userId - {}.", userDto.getId());
+        UserRegistrationRequest request = new UserRegistrationRequest(userDto.getFirstName(), userDto.getLastName(),
+                userDto.getEmail(), "");
+        emailTokenSender.sendEmailVerificationCode(request);
+        log.info("The request to reset the user's password was handled.");
+        return ResponseEntity.status(HttpStatus.OK)
+                .build();
+    }
+
+    @Override
+    @PostMapping(path = "/password/reset/confirm")
+    public ResponseEntity<Void> confirmResetUserPassword(@RequestBody final ConfirmEmailRequest confirmEmailRequest) {
+        log.info("Received email confirmation request to reset password");
+        emailTokenConformer.confirmResetPasswordEmailByCode(
+                new com.zufar.icedlatte.security.dto.ConfirmEmailRequest(confirmEmailRequest.getToken()));
```

Comment:
nit: is it possible to replace FQN with just the method's name?

Reply:
nit: oh, it seems to be conflicting with Open API's class. But what is the purpose of using both `record` and generated class?

Reply:
fixed

Reply:
This change is useful as well, but I was talking about this piece of code 😊

```java
// FQN (fully qualified name)
new com.zufar.icedlatte.security.dto.ConfirmEmailRequest(confirmEmailRequest.getToken()));
// VS simple name
new ConfirmEmailRequest(confirmEmailRequest.getToken()));
```

Reply:
oh no:)
reverted